pkg/cnbutils/bindings/bindings.go

```diff
@@ -16,20 +20,45 @@ type binding struct {
 	Key     string  `json:"key"`
 	Content *string `json:"content,omitempty"`
 	File    *string `json:"file,omitempty"`
+	FromURL *string `json:"fromUrl,omitempty"`
+}
+
+func (b *binding) validate() error {
+	if !validName(b.Key) {
+		return fmt.Errorf("invalid key: '%s'", b.Key)
+	}
+
+	if b.Content == nil && b.File == nil && b.FromURL == nil {
+		return errors.New("one of 'file', 'content' or 'fromUrl' properties must be specified for binding")
+	}
+
+	// Return error if:
+	// 1. Content is set + File or FromURL
+	// 2. File is set + FromURL or Content
+	// 3. FromURL is set + File or Content
+	// 4. Everything is set
+	if (b.Content != nil && (b.File != nil || b.FromURL != nil)) ||
```

Comment:
would it be easier to read (i.e.) no comment needed if we phrase it the other way around, e.g.

```
onlyOneSet := (b.Content != nil && b.File == nil && b.FromURL == nil) ||
   (b.Content == nil && b.File != nil && b.FromURL == nil) ||
   (b.Content == nil && b.File == nil && b.FromURL != nil) {

if !onlyOneSet {
  return error
}
```